File: Mathematical_Algorithms/Java/euclidean_algorithm.java
public class euclidean_algorithm {

    public static int gcd(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }

    public static void main(String[] args) {
        int num1 = 48;
        int num2 = 18;
        System.out.println("Ước chung lớn nhất của " + num1 + " và " + num2 + " là: " + gcd(num1, num2));

        int num3 = 101;
        int num4 = 103;
        System.out.println("Ước chung lớn nhất của " + num3 + " và " + num4 + " là: " + gcd(num3, num4));
    }
}

File: Mathematical_Algorithms/JavaScript/euclidean_algorithm.js
function gcd(a, b) {
    while (b !== 0) {
        let temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

// Ví dụ sử dụng:
const num1 = 48;
const num2 = 18;
console.log(`Ước chung lớn nhất của ${num1} và ${num2} là: ${gcd(num1, num2)}`);

const num3 = 101;
const num4 = 103;
console.log(`Ước chung lớn nhất của ${num3} và ${num4} là: ${gcd(num3, num4)}`);
